src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.managed.cs

```diff
@@ -1191,29 +1175,19 @@ private void ResetReorderingIndex()
 		/// </summary>
 		private protected class Line
 		{
-			public (FrameworkElement container, Uno.UI.IndexPath index)[] Items { get; }
-			public Uno.UI.IndexPath FirstItem { get; }
-			public Uno.UI.IndexPath LastItem { get; }
+			public Uno.UI.IndexPath Item { get; }
 			public int FirstItemFlat { get; }
+			public FrameworkElement View { get; }
 
-			public FrameworkElement FirstView => Items[0].container;
-			public FrameworkElement LastView => Items[Items.Length - 1].container;
-
-			public Line(int firstItemFlat, params (FrameworkElement container, Uno.UI.IndexPath index)[] items)
+			public Line(int firstItemFlat, FrameworkElement container, Uno.UI.IndexPath index)
 			{
-				if (items.Length == 0)
-				{
-					throw new InvalidOperationException("Line must contain at least one view");
-				}
-
-				Items = items;
-				FirstItem = items[0].index;
-				LastItem = items.Last().index;
 				FirstItemFlat = firstItemFlat;
+				View = container;
+				Item = index;
 			}
 
 			public bool Contains(Uno.UI.IndexPath index)
-				=> Items.Any(i => i.index == index);
+				=> Item == index;
```

Comment:
Previously, `Contains` would be allocating a System.Func. Now that it's a simple equality check, it should be more efficient.